nf_core/rocrate.py

```diff
@@ -0,0 +1,300 @@
+                    self.pipeline_obj = Pipeline(str(self.pipeline_dir))
+                    self.pipeline_obj._load()
+                except InvalidGitRepositoryError:
+                    log.error(f"Could not find a git repository in {self.pipeline_dir}")
+                    sys.exit(1)
+                except GitCommandError:
+                    log.error(f"Could not checkout version {self.version}")
+                    sys.exit(1)
+
+        try:
+            self.make_workflow_ro_crate()
+        except Exception as e:
+            log.error(e)
+            sys.exit(1)
+
+        # Save just the JSON metadata file
+        if metadata_fn is not None:
+            log.info(f"Saving metadata file '{metadata_fn}'")
+            # Save the crate to a temporary directory
+            tmpdir = Path(tempfile.mkdtemp(), "wf")
+            self.crate.write(tmpdir)
+            # Now save just the JSON file
+            crate_json_fn = Path(tmpdir, "ro-crate-metadata.json")
+            crate_json_fn.rename(metadata_fn)
+
+        # Save the whole crate zip file
+        if zip_fn is not None:
+            log.info(f"Saving zip file '{zip_fn}'")
+            self.crate.write_zip(zip_fn)
+
+        # Change back to the original directory
+        os.chdir(current_path)
+
+    def make_workflow_ro_crate(self) -> None:
+        """
+        Create an RO Crate for a pipeline
+
+        Args:
+            path (Path): Path to the pipeline directory
+        """
+        if self.pipeline_obj is None:
+            raise ValueError("Pipeline object not loaded")
+
+        # Create the RO Crate object
+        self.crate = rocrate.rocrate.ROCrate()
+
+        # Set language type
+        programming_language = rocrate.model.entity.Entity(
+            self.crate,
+            "#nextflow",
+            properties={
+                "@type": ["ComputerLanguage", "SoftwareApplication"],
+                "name": "Nextflow",
+                "url": "https://www.nextflow.io/",
+                "identifier": "https://www.nextflow.io/",
+                "version": self.pipeline_obj.nf_config.get("manifest.nextflowVersion", ""),
+            },
+        )
+        self.crate.add(programming_language)
+
+        # Conform to RO-Crate 1.1 and workflowhub-ro-crate
+        self.crate.update_jsonld(
+            {
+                "@id": "ro-crate-metadata.json",
+                "conformsTo": [
+                    {"@id": "https://w3id.org/ro/crate/1.1"},
+                    {"@id": "https://w3id.org/workflowhub/workflow-ro-crate/1.0"},
+                ],
+            }
+        )
+
+        # Set main entity file
+        self.set_main_entity("main.nf")
+
+        # add readme as description
+        readme = Path("README.md")
+
+        try:
+            self.crate.description = readme.read_text()
+        except FileNotFoundError:
+            log.error(f"Could not find README.md in {self.pipeline_dir}")
+
+        self.crate.license = "MIT"
```

Comment:
Maybe worth checking for a license file and making sure that it contains the right string, just in case people run this command on non- nf-core pipelines..